### PR TITLE
fix: sort map keys before iteration for deterministic execution

### PR DESCRIPTION
**Pull request by hallazzang**
## Description

Since iteration order of map types is not deterministic, sort map keys before iteration.

### CODE REVIEW

File: x/farming/keeper/reward.go [hunk 1]
```diff
@@ -334,10 +335,20 @@ func (k Keeper) AllocationInfos(ctx sdk.Context) []AllocationInfo {
...
 	}
 
+	// Sort keys for deterministic execution.
+	var planIds []uint64
+	for planId := range plans {
+		planIds = append(planIds, planId)
+	}
+	sort.Slice(planIds, func(i, j int) bool {
+		return planIds[i] < planIds[j]
+	})
```
[the remark this answers is not in the document]
Comment (hallazzang) [reply x/farming/keeper/reward.go:345]: Actually I considered that option once, but chose current code for consistency within added code.
I'll make that change!

**Comment by dongsam** [comment]
we'll test internally whether it's solved to deterministic and approve it.

**Comment by dongsam** [comment]
@hallazzang Could you add commit for fixing `for planID, amt := range planCoins` to deterministic also?

**Comment by dongsam** [comment]
In the TestAppStateDeterminism, I fixed that the seed was set to 1 and the test was not done properly, after that, the liquidity `MatchOrders` seems to have panic, so could you please check it?

File: x/farming/keeper/reward.go [hunk 2]
```diff
@@ -370,7 +375,15 @@ func (k Keeper) AllocationInfos(ctx sdk.Context) []AllocationInfo {
 	// In this step, we check if farming pools have sufficient balance for allocations.
 	// If not, we don't allocate rewards from that farming pool for this epoch.
 	var allocInfos []AllocationInfo
-	for farmingPool, planCoins := range allocCoins {
+	// Sort map keys for deterministic execution.
+	var farmingPools []string
+	for farmingPool := range allocCoins {
+		farmingPools = append(farmingPools, farmingPool)
+	}
+	sort.Strings(farmingPools)
```
Comment (kogisin) [inline x/farming/keeper/reward.go:383]: This is me nitpicking to have space between two different logics. For better readability, I suggest to have space in the following line. Also, for the same reason I think `var allocInfos []AllocationInfo` can move down to line 385-386 (after `sort.Strings(farmingPools)`).
```suggestion
	sort.Strings(farmingPools)
	
```

File: x/farming/keeper/reward.go [hunk 3]
```diff
@@ -471,7 +484,15 @@ func (k Keeper) AllocateRewards(ctx sdk.Context) error {
 
 	// For each staking coin denom in the table, increase cumulative unit rewards
 	// and increment current epoch number by 1.
-	for stakingCoinDenom, unitRewards := range unitRewardsByDenom {
+	// Sort keys for deterministic execution.
+	var denoms []string
+	for denom := range unitRewardsByDenom {
+		denoms = append(denoms, denom)
+	}
+	sort.Strings(denoms)
```
Comment (kogisin) [inline x/farming/keeper/reward.go:492]: Same comment as above goes here in this line.
```suggestion
	sort.Strings(denoms)
	
```

**Review by kogisin — CHANGES_REQUESTED** [review]
Simulation [test-sim-nondeterminism](https://github.com/cosmosquad-labs/squad/runs/5617805420?check_suite_focus=true#logs) is failing due to negative coin amount.

File: x/farming/keeper/reward.go (hunk 2 above)
Comment (hallazzang) [reply x/farming/keeper/reward.go:380]: I updated the code style, thanks.

File: x/farming/keeper/reward.go (hunk 3 above)
Comment (hallazzang) [reply x/farming/keeper/reward.go:500]: I updated the code style, thanks.

**Review by kogisin — APPROVED** [review]

**Review by kogisin — COMMENTED** [review]
Simulation test is still failing.

**Review by dongsam — APPROVED** [review]

**Pull request merged**